### PR TITLE
modified ROSThreadImage (and affected files) to work without ROSThread

### PR DESCRIPTION
**Pull request by sarimabbas**
So i think i've got ROSThreadImage working without ROSThread. I went ahead and modified @akrzywosz 
 test_ros_thread_image.cpp and the robot_perception/cartesian_estimator.cpp with lockguards and thread close flags

### CODE REVIEW

File: baxter_collaboration_lib/include/robot_utils/ros_thread_image.h [hunk 1]
```diff
@@ -7,29 +7,36 @@
 #include <opencv2/imgproc/imgproc.hpp>
 #include <opencv2/highgui/highgui.hpp>
 
-#include "robot_utils/ros_thread.h"
+// #include "robot_utils/ros_thread.h"
```
Comment (alecive) [inline baxter_collaboration_lib/include/robot_utils/ros_thread_image.h]: cleanup

File: baxter_collaboration_lib/include/robot_utils/ros_thread_image.h [hunk 2]
```diff
@@ -7,29 +7,36 @@
...
  *          to be overwritten by its children. Useful to to visual processing.
  */
-class ROSThreadImage : public ROSThread
+class ROSThreadImage
 {
 protected:
     ros::NodeHandle _n;
 
 private:
     std::string    _name;
```
Comment (alecive) [inline baxter_collaboration_lib/include/robot_utils/ros_thread_image.h]: remove underscore here

File: baxter_collaboration_lib/include/robot_utils/ros_thread_image.h [hunk 3]
```diff
@@ -7,29 +7,36 @@
...
+    std::thread     image_thread;
```
Comment (alecive) [inline baxter_collaboration_lib/include/robot_utils/ros_thread_image.h]: call it img_thread

File: baxter_collaboration_lib/include/robot_utils/ros_thread_image.h (hunk 2 above)
Comment (alecive) [reply baxter_collaboration_lib/include/robot_utils/ros_thread_image.h]: (and add the underscore in the constructor [the input argument])

File: baxter_collaboration_lib/src/robot_utils/ros_thread_image.cpp [hunk 4]
```diff
@@ -4,25 +4,40 @@
...
-    pthread_mutexattr_init(&_mutex_attr);
-    pthread_mutexattr_settype(&_mutex_attr, PTHREAD_MUTEX_RECURSIVE_NP);
-    pthread_mutex_init(&_mutex_img, &_mutex_attr);
 
     _img_sub = _img_trp.subscribe("/"+getName()+"/image", // "/cameras/right_hand_camera/image",
                            SUBSCRIBER_BUFFER, &ROSThreadImage::imageCb, this);
 
     spinner.start();
-    startInternalThread();
+    // startInternalThread();
```
Comment (alecive) [inline baxter_collaboration_lib/src/robot_utils/ros_thread_image.cpp]: cleanup

**Review by alecive — CHANGES_REQUESTED** [review]

File: baxter_collaboration_lib/include/robot_utils/ros_thread_image.h (hunk 2 above)
Comment (sarimabbas) [reply baxter_collaboration_lib/include/robot_utils/ros_thread_image.h]: are you referring to `ros::NodeHandle _n` and `std::string _name` both?

**Comment by sarimabbas** [comment]
removed underscore from `std::string _name` but not from `ros::NodeHandle _n` because `cartesian_estimator_hsv.cpp` and `cartesian_estimator.cpp` seem to rely on `_n` extensively.

**Review by alecive — APPROVED** [review]

**Pull request merged**